src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/autocrafters/AbstractAutoCrafter.java

```diff
@@ -414,8 +434,8 @@ public boolean craft(@Nonnull Inventory inv, @Nonnull AbstractRecipe recipe) {
             return false;
         }
 
-        // Check if we have an empty slot
-        if (inv.firstEmpty() != -1) {
+        // Check if the inventory have any available space
+        if (inventoryAvailable(inv, recipe.getResult())) {
```

Comment:
```suggestion
        if (InvUtils.fits(inv, recipe.getResult())) {
```

Reply:
Ohhh ~ Okay! Thank you. Sorry I'm just kinda noob I didn't know how to use this method. Very noice.